### PR TITLE
[fm] Harden SitrepGuardedInsert dummy-resource test coverage.

### PR DESCRIPTION
**Pull request by mergeconflict**
The existing tests for `SitrepGuardedInsert` (from #10564) all ran against a single-sitrep history, which can't distinguish "the guard compares against the current sitrep" from "the guard matches any sitrep in history". This change adds a new parent sitrep in a couple places to provide better coverage for the guard's behavior.

While we're at it, we also add a brand new test ensuring that we fail closed in the unexpected case where sitrep history is empty.

### CODE REVIEW

**Comment by smklein** [comment]
PR description should probably reference https://github.com/oxidecomputer/omicron/pull/10564 ; #10532 is your PR that you closed

File: nexus/db-queries/src/db/sitrep_guard.rs [hunk 1]
```diff
@@ -618,8 +633,12 @@ mod tests {
 
         assert_matches!(outcome, SitrepGuardedInsertOutcome::AlreadyExists);
         // The resource row was not inserted; the seeded marker is unchanged.
+        // If the marker INSERT were ever flipped to an UPSERT (or lost its
+        // `ON CONFLICT DO NOTHING`), this would read Some(2), breaking the
+        // invariant that markers record the generation at which the resource
+        // was originally created.
```
Comment (smklein) [inline nexus/db-queries/src/db/sitrep_guard.rs]: Would this actually be `Some(2)` from that change alone? `prior_marker_guard` would abort the whole statement with a sentinel error, and the CTE is atomic, so `new_marker` would never get executed regardless of the `ON CONFLICT` clause`.

File: nexus/db-queries/src/db/sitrep_guard.rs [hunk 2]
```diff
@@ -686,4 +713,32 @@ mod tests {
         db.terminate().await;
         logctx.cleanup_successful();
     }
+
+    // With no sitrep in history at all, there is no current generation to
+    // match, so the guarded insert must fail closed: no resource row and no
+    // marker row may be written. We deliberately do not assert *which*
+    // outcome is reported -- this situation is unreachable via fm_rendezvous,
```
Comment (smklein) [inline nexus/db-queries/src/db/sitrep_guard.rs:721]: `run_guarded_insert` (called below) does unwrap the result from `execute_async`, so we do need one of the `SitrepGuardedInsertOutcome` values to be returned. I think it's *probably* returning `StaleSitrep`?

Why not validate that's being returned? The predictability here seems worthwhile asserting, even if we have the comment caveat "This situation is a bit weird".

(also I think this is just making https://github.com/oxidecomputer/omicron/pull/10564#discussion_r3375413267 explicit)

**Review by smklein — APPROVED** [review]

File: nexus/db-queries/src/db/sitrep_guard.rs (hunk 2 above)
Comment (mergeconflict) [reply nexus/db-queries/src/db/sitrep_guard.rs:721]: I don't want to specify a behavioral contract that this returns `StaleSitrep` in the no-sitreps case. The only guarantee I want to make here is that nothing bad happens.

File: nexus/db-queries/src/db/sitrep_guard.rs (hunk 1 above)
Comment (mergeconflict) [reply nexus/db-queries/src/db/sitrep_guard.rs]: Yeah, the comment was lying, fixed in b439e05.

File: nexus/db-queries/src/db/sitrep_guard.rs (hunk 2 above)
Comment (smklein) [reply nexus/db-queries/src/db/sitrep_guard.rs:721]: my point was mostly that "we probably want to know if this changes", not "this specific value must be correct"
